sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1813,6 +1813,26 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("SPARK-30104: global temp db is used as a table name under v2 catalog") {
+    val globalTempDB = spark.sessionState.conf.getConf(StaticSQLConf.GLOBAL_TEMP_DATABASE)
+    val t = s"testcat.$globalTempDB"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (id bigint, data string) USING foo")
+      sql("USE testcat")
+      // The following should not throw AnalysisException, but should use `testcat.$globalTempDB`.
+      sql(s"DESCRIBE TABLE $globalTempDB")
+    }
+  }
+
+  test("table name same as catalog can be used") {
+    withTable("testcat.testcat") {
+      sql(s"CREATE TABLE testcat.testcat (id bigint, data string) USING foo")
+      sql("USE testcat")
+      // The following should not throw AnalysisException.
+      sql(s"DESCRIBE TABLE testcat")
+    }
```

Comment:
I just noticed that the following would fail: 
```
sql(s"CREATE TABLE testcat.testcat.testcat (id bigint, data string) USING foo")
sql("USE testcat")
sql(s"DESCRIBE TABLE testcat.testcat") // First part of 'testcat' is resolved to a catalog, not a namespace.
```

Reply:
This is expected. We can add many special cases to the resolution policy, but it's better to keep the policy simple.

According to the original design, the first name part should be resolved to catalog first, even if it ends up with table not found.

I think the single-part name should be the only special case, as table name can't be an empty string array.

Reply:
Got it, thanks! And the user can always use the fully qualified name to disambiguate.